opentelemetry-sdk/src/logs/log_emitter.rs

```diff
@@ -246,7 +247,11 @@ impl opentelemetry::logs::Logger for Logger {
     type LogRecord = LogRecord;
 
     fn create_log_record(&self) -> Self::LogRecord {
-        LogRecord::default()
+        // Reserve attributes memory for perf optimization. This may change in future.
+        LogRecord {
+            attributes: Some(Vec::with_capacity(PREALLOCATED_ATTRIBUTE_CAPACITY)),
+            ..Default::default()
```

Comment:
Just curious, what is the `..Default` for?

Reply:
`..Default::default()` means that all the remaining fields of the struct that are not explicitly set will be initialized to their default values. From `C++` prospective, this can be thought of as using constructor overloading, where one constructor initializes specific fields, and another constructor (the default constructor) initializes the rest. `Rust` doesn't have concept of function overloading, so this is the pattern used.